From e1a8d1490be417d763410639b68b0579310ebbe1 Mon Sep 17 00:00:00 2001
From: Juri Lelli <juri.lelli@gmail.com>
Date: Mon, 5 Nov 2012 09:50:44 -0800
Subject: [PATCH 43/48] rt_handler sleeping function in atomic context fixed

Switch the order of two operation inside sched_rt_handler to
resolve a possible - sleeping function in atomic context - BUG.
---
 kernel/sched/core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index adb8d2c..b1eff9d 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -7992,9 +7992,9 @@ int sched_rt_handler(struct ctl_table *table, int write,
 	ret = proc_dointvec(table, write, buffer, lenp, ppos);
 
 	if (!ret && write) {
+		ret = sched_rt_global_constraints();
 		raw_spin_lock_irqsave(&def_dl_bandwidth.dl_runtime_lock,
 				      flags);
-		ret = sched_rt_global_constraints();
 		if (ret) {
 			sysctl_sched_rt_period = old_period;
 			sysctl_sched_rt_runtime = old_runtime;
-- 
1.7.9.5

